target/linux/ramips/mt7621/base-files/lib/upgrade/platform.sh

```diff
@@ -33,6 +33,15 @@ platform_do_upgrade() {
 			fw_setenv --lock / bootImage 0 || exit 1
 		fi
 		;;
+	h3c,tx1806)
+		bootargs="$(fw_printenv -n bootargs 2>/dev/null)"
+		if [ -n "$bootargs" ]; then
+			fw_setenv bakargs "$bootargs" && fw_setenv bootargs || exit 1
+		fi
```

Comment:
`bootargs=console=ttyS0,115200 rootfstype=squashfs loglevel=3 root=/dev/mtdblock7/8/9/..`
rename bootargs to ensure kernel get the correct partition and full log support.

Reply:
Err, can this be resolved by adding `bootargs-append = " root="` in the DTS "chosen" node?
Even if not, we should access UART during installation anyway, I believe it's better to handle it once in the U-Boot shell directly, rather than poking u-boot-env for every sysupgrade.

Reply:
I updated the flash instructions and now we can drop it. When "bootargs" is removed u-boot will use default arguments and uart will work as usual.